src/Servers/HttpSys/src/MessagePump.cs

```diff
@@ -234,7 +234,9 @@ private async void ProcessRequestAsync(object requestContextObj)
                     {
                         // We haven't sent a response yet, try to send a 500 Internal Server Error
                         requestContext.Response.Headers.IsReadOnly = false;
+                        requestContext.Response.Trailers.IsReadOnly = false;
```

Comment:
nit: Why do you need to set IsReadOnly?

Reply:
This code resets the response to generate an internal error. E.g. an error happened while serializing the headers (uncommon). You can't call Clear when the collection is in read-only mode.

Reply:
Ah, calling clear. Coolio.